core/assertion.ts

```diff
@@ -78,13 +78,17 @@ export class AssertionContext {
     this.assertion = assertion;
   }
 
-  public ref(name: string) {
+  public ref(reference: string | IResolvable) {
+    const name =
+      typeof reference === "string" || typeof reference === "undefined"
+        ? reference
+        : (reference as IResolvable).schema + "." + (reference as IResolvable).name;
```

Comment:
you *shouldn't* need to type-assert `reference` as `IResolvable`: typescript should be clever enough to know that if `reference` is not a string (as in this case), it must be `IResolvable`

(typescript's type system is crazy good)

Reply:
Done